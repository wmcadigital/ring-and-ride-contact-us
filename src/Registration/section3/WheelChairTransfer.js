import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getRequirementsSectionName from "./getRequirementsSectionName";
import { required } from "../../common/validation";

const WheelChairTransfer = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const error = stateApi.submitFailed
    ? stateApi.errors?.wheelChairTransfer
    : null;

  let wheelChairType = "";

  if (formValues.mobilityAids.manualWheelchair === true) {
    wheelChairType = "manual";
  } else {
    wheelChairType = "powered";
  }

  const question =
    formValues["registerForYourself"] === "yes"
      ? `Are you able to transfer yourself from your ${wheelChairType} wheelchair?`
      : `Is ${formValues["firstName"]} able to transfer themselves from their ${wheelChairType} wheelchair?`;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 3 of 3"
        sectionName={getRequirementsSectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="wheelChairTransfer"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="wheelChairTransfer"
        />
      </RadioGroup>
    </FormSection>
  );
};

export default WheelChairTransfer;
