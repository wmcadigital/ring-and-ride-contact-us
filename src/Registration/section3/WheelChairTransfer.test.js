import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import WheelChairTransfer from "./WheelChairTransfer";

jest.mock("react-final-form");

describe("WheelChairTransfer", () => {
  it("individual is filling out form for themselves - manual wheel chair", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
        mobilityAids: {
          manualWheelchair: true,
        },
      },
    }));
    const renderer = createRenderer();

    renderer.render(<WheelChairTransfer />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else - powered wheel chair", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "no",
        firstName: "Test",
        lastName: "Testerson",
        mobilityAids: {
          powerWheelchair: true,
        },
      },
    }));
    const renderer = createRenderer();

    renderer.render(<WheelChairTransfer />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
        mobilityAids: {
          manualWheelchair: true,
        },
      },
      submitFailed: true,
      errors: { wheelChairTransfer: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<WheelChairTransfer />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
