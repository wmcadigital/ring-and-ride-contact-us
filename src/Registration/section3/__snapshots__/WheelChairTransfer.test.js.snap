// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WheelChairTransfer individual is filling out form for someone else - powered wheel chair 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Applicant requirements"
    sectionPosition="Section 3 of 3"
  />
  <Question
    text="Is Test able to transfer themselves from their powered wheelchair?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`WheelChairTransfer individual is filling out form for themselves - manual wheel chair 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Your requirements"
    sectionPosition="Section 3 of 3"
  />
  <Question
    text="Are you able to transfer yourself from your manual wheelchair?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`WheelChairTransfer renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Your requirements"
    sectionPosition="Section 3 of 3"
  />
  <Question
    text="Are you able to transfer yourself from your manual wheelchair?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="wheelChairTransfer"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
